simde/x86/sse4.2.h

```diff
@@ -241,6 +241,57 @@ simde_mm_cmpistrz_16_(simde__m128i b) {
   #define _mm_cmpistrz(a, b, imm8) simde_mm_cmpistrz(a, b, imm8)
 #endif
 
+SIMDE_FUNCTION_ATTRIBUTES
+int64_t bitreflection(int64_t n) {
+  int64_t rev = 0;
+  while(n > 0){
+    rev <<= 1;
+    if(n&1)
+      rev^=1;
+    n >>= 1;
+  }
+  return rev;
+}
+
+SIMDE_FUNCTION_ATTRIBUTES
+int64_t mod2div(int64_t dividend, int64_t divisor){
+  int64_t remainder = dividend;
+  int64_t sdata = floor(log2(dividend))+1;
```

Comment:
Instead of using libm directly, you should use the math functions in simde/simde-math.h.  IIRC you'll need to add a `simde_math_log2`, but `simde_math_floor` already exists.